src/Components/v2/CarouselV2.tsx

```diff
@@ -0,0 +1,265 @@
+import { Box, ChevronIcon, Flex } from "@artsy/palette"
+import React, { ReactNode, useEffect, useRef, useState } from "react"
+import styled from "styled-components"
+import { left, LeftProps, right, RightProps } from "styled-system"
+import { Media } from "Utils/Responsive"
+
+export interface FlickitySettings {
+  draggable?: boolean
+  freeScroll?: boolean
+  wrapAround?: boolean
+  groupCells?: boolean | number | string
+  autoPlay?: boolean | number
+  fullscreen?: boolean
+  fade?: boolean
+  adaptiveHeight?: boolean
+  watchCSS?: boolean
+  hash?: boolean
+  dragThreshold?: number
+  friction?: number
+  selectedAttraction?: number
+  freeScrollFriction?: number
+  imagesLoaded?: boolean
+  lazyLoad?: boolean | number
+  bgLazyLoad?: boolean | number
+  initialIndex?: number
+  accessibility?: boolean
+  setGallerySize?: boolean
+  resize?: boolean
+  cellAlign?: "left" | "center" | "right"
+  contain?: boolean
+  percentPosition?: boolean
+  rightToLeft?: boolean
+  prevNextButtons?: boolean
+  pageDots?: boolean
+}
+
+type Arrow = (
+  props: {
+    Arrow: React.ReactType
+    getFlickity: any
+  } & Props
+) => React.ReactNode
+
+interface Props {
+  height?: number
+  data: object[] // This is designed to handle any shape of data passed, as long as its an array
+  render: (slide) => ReactNode
+  renderLeftArrow?: Arrow
+  renderRightArrow?: Arrow
+  onArrowClick?: () => void
+  settings?: FlickitySettings
+}
+
+export class Carousel extends React.Component<Props> {
+  static defaultProps = {
+    height: 300,
+  }
+
+  render() {
+    return (
+      <>
+        <Media at="xs">
+          <SmallCarousel {...this.props} />
+        </Media>
+        <Media greaterThan="xs">
+          <LargeCarousel {...this.props} />
+        </Media>
+      </>
+    )
+  }
+}
+
+/*
+ * Returns null if no window for SSR
+ */
+const Flickity =
+  typeof window !== "undefined"
+    ? require("react-flickity-component")
+    : () => null
```

Comment:
@kierangillen - can you post a screen-cap of this new work running (linked) in force with JS disabled, showing that the layout isn't broken during the SSR pass?

(Also, we need a screen cap of it running regularly in the description, too, along with some content from the now-closed RFC describing the _why_ around why this switch was necessary).